File: Programmers/[12899] 124 나라의 숫자/ONE/12899.java
class Solution {
    public String solution(int n) {
        int num = n;
        int[] matching = {4, 1, 2};
        StringBuilder answer = new StringBuilder();

        while (num > 0) {
            answer.insert(0, matching[num % 3]);
            num /= 3;
            if (answer.charAt(0) == '4')
                num--;
        }
        return answer.toString();
    }
}

File: Programmers/[12899] 124 나라의 숫자/pinkishincoloragain/12899.py
def solution(n):
    answer = ""
    res = n
    while res > 0:
        if res % 3 != 0:
            answer += str(res % 3)
        if res % 3 == 0:
            res = res//3 -1
            answer+="4"
        else:
            res = res // 3
            
    return answer[::-1]

